### PR TITLE
Adds an after success run for e2e pilot

### PR DESCRIPTION
**Pull request by sebastienvas**
-->
```release-note
none
```

### CODE REVIEW

**Review by guptasu — APPROVED** [review]

File: prow/config.yaml [hunk 1]
```diff
@@ -254,6 +254,51 @@ presubmits:
         hostPath:
           path: /mnt/disks/ssd0
     run_after_success:
+    - name: istio-pilot-e2e
+      agent: kubernetes
+      context: e2e-smoke.sh
```
Comment (yutongz) [inline prow/config.yaml]: Change this?

File: prow/config.yaml [hunk 2]
```diff
@@ -254,6 +254,51 @@ presubmits:
...
+            hostPort: 9998
+          resources:
+            requests:
+              memory: "512Mi"
+              cpu: "500m"
+            limits:
+              memory: "20Gi"
+              cpu: "5000m"
+        nodeSelector:
+          cloud.google.com/gke-nodepool: build-pool
```
Comment (yutongz) [inline prow/config.yaml:294]: You need to use "new-cluster-pool", which doesn't have kubeconfig mounted

**Review by yutongz — CHANGES_REQUESTED** [review]

File: prow/config.yaml (hunk 1 above)
Comment (sebastienvas) [reply prow/config.yaml]: Done

File: prow/config.yaml (hunk 2 above)
Comment (sebastienvas) [reply prow/config.yaml:294]: The kubeconfig is mounted on the pod not on the pool.

**Review by yutongz — APPROVED** [review]

**Pull request merged**